python/foglamp/tasks/purge/purge.py

```diff
@@ -109,31 +110,32 @@ def purge_data(self, config):
 
                 result = self._readings_storage.purge(age=config['age']['value'], sent_id=last_id, flag=flag)
 
-                if "message" in result.keys() and "409 Conflict" in result["message"]:
-                    self._logger.error("Purge failed: %s", result["message"])
-                else:
-                    total_count = result['readings']
-                    total_rows_removed = result['removed']
-                    unsent_rows_removed = result['unsentPurged']
-                    unsent_retained = result['unsentRetained']
+                total_count = result['readings']
+                total_rows_removed = result['removed']
+                unsent_rows_removed = result['unsentPurged']
+                unsent_retained = result['unsentRetained']
         except ValueError:
             self._logger.error("Configuration item age {} should be integer!".format(config['age']['value']))
             raise ValueError
+
+        except StorageServerError:
+            pass
```

Comment:
see error content and retry-able ?!

At least  log self._logger.error("Failed to purge,  %s", result["message"])